lib/src/license_detection/license.dart

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:meta/meta.dart';
+import 'package:pana/src/license_detection/tokenizer.dart';
+
+@sealed
+class License {
+  /// Name of the license, is empty in case of unknown license.
+  final String identifier;
+
+  /// Original text from the license file.
+  final String content;
+
+  /// Normalized [Token]s created from the original text.
+  final List<Token> tokens;
+
+  /// A map of tokens and their count.
+  final Map<String, int> occurences;
+
+  License._(this.content, this.tokens, this.occurences, this.identifier);
+
+  factory License.parse(String identifier, String content) {
+    final tokens = tokenize(content);
+    final table = generateFrequencyTable(tokens);
+    return License._(content, tokens, table, identifier);
+  }
+}
+
+/// Contains deatils regarding the results of corpus license match with unknwown text.
+@sealed
+class LicenseMatch {
+  /// Sequence of tokens that were found in the unknown
+  ///  text that matched to tokens in any of the corpus license.
+  final List<Token> tokens;
+
+  /// Confidence score of the detected license.
+  final double confidence;
+
+  /// SPDX license which matched with input.
+  final License license;
+
+  LicenseMatch(this.tokens, this.confidence, this.license);
+}
+
+/// Genearates a frequency table for the give list of tokens.
+@visibleForTesting
+Map<String, int> generateFrequencyTable(List<Token> tokens) {
+  var table = <String, int>{};
+
+  for (var token in tokens) {
+    table[token.value] = table.putIfAbsent(token.value, () => 0) + 1;
+  }
+
+  return table;
+}
+
+/// Creates [License] instances for all the corpus licenses.
```

Comment:
How is directories searched? what is the expected format? what is the filename format?

Maybe, this is correct?
```suggestion
/// Load a list of [License] from license files found in [directories].
///
/// The [directories] are not searched recursively, and license files are
/// expected to be plaintext with a filename on the form `<spdx-identifier.txt>`.
```